drivers/arduino-hid.c

```diff
@@ -0,0 +1,145 @@
+/* arduino-hid.c - subdriver to monitor Arduino USB/HID devices with NUT
+ *
+ *  Copyright (C)
+ *  2003 - 2012	Arnaud Quette <ArnaudQuette@Eaton.com>
+ *  2005 - 2006	Peter Selinger <selinger@users.sourceforge.net>
+ *  2008 - 2009	Arjen de Korte <adkorte-guest@alioth.debian.org>
+ *  2013 Charles Lepple <clepple+nut@gmail.com>
+ *
+ *  Note: this subdriver was initially generated as a "stub" by the
+ *  gen-usbhid-subdriver script. It must be customized.
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+
+#include "usbhid-ups.h"
+#include "arduino-hid.h"
+#include "main.h"	/* for getval() */
+#include "usb-common.h"
+
+#define ARDUINO_HID_VERSION	"Arduino HID 0.1"
+/* FIXME: experimental flag to be put in upsdrv_info */
+
+/* Arduino */
+#define ARDUINO_VENDORID	0x2341
+#define ARDUINO_VENDORID2	0x2A03
+
+/* USB IDs device table */
+static usb_device_id_t arduino_usb_device_table[] = {
+	/* Arduino Leonardo, Leonardo ETH and Pro Micro*/
+        { USB_DEVICE(ARDUINO_VENDORID, 0x0036), NULL },
+	{ USB_DEVICE(ARDUINO_VENDORID, 0x8036), NULL },
+        { USB_DEVICE(ARDUINO_VENDORID2, 0x0036), NULL },
+	{ USB_DEVICE(ARDUINO_VENDORID2, 0x8036), NULL },
+        { USB_DEVICE(ARDUINO_VENDORID2, 0x0040), NULL },
+        { USB_DEVICE(ARDUINO_VENDORID2, 0x8040), NULL },
```

Comment:
Cosmetic: mismatch of tabs and spaces to indent (also in other parts of the PRed code)

Reply:
@jimklimov will fix in the next commit. I think this is because the file was generated first and then fixed through NetBeans IDE, which converts tabs to spaces automatically.

Reply:
fixed